### PR TITLE
chore:  Drop iOS 15 support - WPB-11374

### PR DESCRIPTION
**Pull request by samwyndham**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11374" title="WPB-11374" target="_blank">TaskWPB-11374</a>  Drop support iOS 15
  </td></table>
  <br />
 

### Issue

This PR drops support for `iOS 15` setting the minimum deployment target to `iOS 16.0`.

This has added a number of deprecation warnings to our project. As fixing these will require some effort, I have created linked follow up tickets rather than fixing them in this PR (I did fix one in this PR which was just a rename)

### Testing

Run automated tests

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios-system/WireSystem.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -835,7 +835,6 @@
 				GENERATE_INFOPLIST_FILE = YES;
 				INFOPLIST_KEY_NSHumanReadableCopyright = "Copyright © 2024 Wire Swiss GmbH. All rights reserved.";
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
-				IPHONEOS_DEPLOYMENT_TARGET = 15.0;
```
Comment (samwyndham) [inline wire-ios-system/WireSystem.xcodeproj/project.pbxproj:838]: Here and similar I remove unnecessarily overriding the `IPHONEOS_DEPLOYMENT_TARGET` values which have been set in config files, rather than updating this value

File: wire-ios/Wire-iOS/Sources/UserInterface/Location/MapViewController.swift [hunk 2]
```diff
@@ -40,7 +40,7 @@ final class MapViewController: UIViewController {
     let mapView = MKMapView()
     weak var delegate: MapViewControllerDelegate?
     private let pointAnnotation = MKPointAnnotation()
-    private lazy var annotationView: MKPinAnnotationView = MKPinAnnotationView(
```
Comment (samwyndham) [inline wire-ios/Wire-iOS/Sources/UserInterface/Location/MapViewController.swift:43]: This was changed to fix a deprecation warning (rename).

File: wire-ios/WireCommonComponents/FileMetaDataGenerator/FileMetaDataGenerator.swift [hunk 3]
```diff
@@ -42,7 +42,7 @@ public final class FileMetaDataGenerator: FileMetaDataGeneratorProtocol {
 
         let thumbnail: Data?
         do {
-            thumbnail = try await previewGenerator.generatePreviewForFile(at: url)
```
Comment (samwyndham) [inline wire-ios/WireCommonComponents/FileMetaDataGenerator/FileMetaDataGenerator.swift:45]: The `await` was not necessary here - there was a build warning. This change is unrelated to dropping iOS 15

File: wire-ios/WireCommonComponents/FileMetaDataGenerator/FileMetaDataGenerator.swift
Comment (netbe) [inline wire-ios/WireCommonComponents/FileMetaDataGenerator/FileMetaDataGenerator.swift:1]: @samwyndham are the warnings in this file part of this PR or part of related tickets?

File: SourceryPlugin/Package.swift [hunk 4]
```diff
@@ -4,7 +4,7 @@ import PackageDescription
 
 let package = Package(
     name: "SourceryPlugin",
-    platforms: [.iOS(.v15), .macOS(.v12)],
+    platforms: [.iOS(.v16), .macOS(.v12)],
```
Comment (netbe) [inline SourceryPlugin/Package.swift:7]: out of scope but it would be great if we had a way to specify the minimum version once in the workspace and it to be read in all packages and projects

**Review by netbe — COMMENTED** [review]
Looks good, just added two comments before approving

File: wire-ios/WireCommonComponents/FileMetaDataGenerator/FileMetaDataGenerator.swift
Comment (samwyndham) [reply wire-ios/WireCommonComponents/FileMetaDataGenerator/FileMetaDataGenerator.swift:1]: @netbe I've create 3 separate tickets for fixing warnings in this file and others generated by dropping iOS 15. These are added to the sprint and assigned to me. The warnings in this file are covered by https://wearezeta.atlassian.net/browse/WPB-11499. I'm doing these as separate tickets because fixing them requires a fair amount of work - e.g. migrating to `async` methods

File: SourceryPlugin/Package.swift (hunk 4 above)
Comment (samwyndham) [reply SourceryPlugin/Package.swift:7]: To the best of my knowledge and after a little research it is not possible to import anything into these package manifests. However we could use code generation to create some reusable code in these manifest files such as minimum version. I think it would only be worth it if we find there is more code being reused, otherwise the abstraction could complicate things.

**Review by netbe — APPROVED** [review]

**Review by KaterinaWire — APPROVED** [review]

**Comment by markusfassbender** [comment]
Yay 🎉👋
